client_side/.infrastructure/infrastructure.sh

```diff
@@ -0,0 +1,201 @@
+##############################################################################
+# This file contains utility functions for use by the interface:
+# - Creating the mock file system for the user
+# - Timing each task
+# - Determining which task and treatment ordering for the current experiment
+# - Determine the next task to move onto and whether the experiment is over
+# - Verify the output of a task
+##############################################################################
+
+# Converts a numeric ASCII value to a character.
+chr() {
+  [ "$1" -lt 256 ] || return 1
```

Comment:
I think it would be better to crash here (certainly during testing!), rather than silently returning an incorrect value.

Reply:
The actual "return" value of `chr` is what it prints not the `return` statement. `return` here is dictating its return code. There are now comments documenting this behavior for both `chr` and `ord`.